porter/v1/cluster_control_plane.proto

```diff
@@ -221,22 +166,114 @@ service ClusterControlPlaneService {
   // This should ultimately be wrapped into UpdateContract
   rpc DeleteCluster(DeleteClusterRequest) returns (DeleteClusterResponse) {}
 
-  // ECRTokenForRegistry returns a docker-compatible token for accessing a given ECR registry
-  rpc ECRTokenForRegistry(ECRTokenForRegistryRequest) returns (ECRTokenForRegistryResponse) {}
-
-  // AssumeRoleCredentials should be used vary sparingly, and ONLY for replacing AWS Integrations which have no workaround on the Porter API.
-  // This endpoint returns temporary AWS credentials for a given AWS Account ID, and should not be expanded further to allow specifc role selection without being tied to a project and cluster
-  rpc AssumeRoleCredentials(AssumeRoleCredentialsRequest) returns (AssumeRoleCredentialsResponse) {}
-
   // TokenForRegistry returns a token for accessing a given registry
   rpc TokenForRegistry(TokenForRegistryRequest) returns (TokenForRegistryResponse) {}
 
-  // DockerConfigFileForRegistry returns a stringified config.json for accessing a given registry
-  rpc DockerConfigFileForRegistry(DockerConfigFileForRegistryRequest) returns (DockerConfigFileForRegistryResponse) {}
-
   // ListRepositoriesForRegistry lists the repositories for a given registry, provided it is in the scope of the project id
   rpc ListRepositoriesForRegistry(ListRepositoriesForRegistryRequest) returns (ListRepositoriesForRegistryResponse) {}
 
   // ListImagesForRepository lists the repositories for a given registry, provided it is in the scope of the project id
   rpc ListImagesForRepository(ListImagesForRepositoryRequest) returns (ListImagesForRepositoryResponse) {}
+
+  // DockerConfigFileForRegistry returns a stringified config.json for accessing a given registry.
+  // Deprecated. Use TokenForRegistry instead.
+  rpc DockerConfigFileForRegistry(DockerConfigFileForRegistryRequest) returns (DockerConfigFileForRegistryResponse) {
+    option deprecated = true;
+  }
+
+  // ECRTokenForRegistry returns a docker-compatible token for accessing a given ECR registry. Deprecated. Use TokenForRegistry instead.
+  rpc ECRTokenForRegistry(ECRTokenForRegistryRequest) returns (ECRTokenForRegistryResponse) {
```

Comment:
This cannot be deleted before TokenForRegistry supports AWS and we edit `RegistryGetECRTokenHandler` to call that endpoint

Reply:
Will add to the comment. Plan here is to deprecated everything now, then update the endpoints after GKE is done.